datasetinsights/estimators/faster_rcnn.py

```diff
@@ -80,59 +141,60 @@ def __init__(
             box_score_thresh=box_score_thresh,
         )
         self.model_without_ddp = self.model
-        self.gpu = gpu
-        self.rank = rank
         self.sync_metrics = config.get("synchronize_metrics", True)
-        logger.info(f"gpu: {self.gpu}, rank: {self.rank}")
-
-        self.checkpointer = checkpointer
-        checkpoint_file = config.get("checkpoint_file", const.NULL_STRING)
-        if checkpoint_file != const.NULL_STRING:
-            checkpointer.load(self, config.checkpoint_file)
-
+        self.data_root = data_root
         self.metrics = {}
         for metric_key, metric in config.metrics.items():
             self.metrics[metric_key] = EvaluationMetric.create(
                 metric.name, **metric.args
             )
         self.model.to(self.device)
 
-        if self.config.system.distributed:
+        if self.distributed:
             self.model = torch.nn.parallel.DistributedDataParallel(
                 self.model, device_ids=[self.gpu]
             )
             self.model_without_ddp = self.model.module
 
+        self.checkpoint_file = self.config.get(
+            "checkpoint_file", const.NULL_STRING
+        )
+        if self.checkpoint_file != const.NULL_STRING:
+            self.checkpointer.load(self, self.config.checkpoint_file)
+
     def train(self, **kwargs):
         """start training, save trained model per epoch."""
         config = self.config
-        train_dataset = create_dataset(config, TRAIN)
-        val_dataset = create_dataset(config, VAL)
+        train_dataset = create_dataset(config, self.data_root, TRAIN)
+        val_dataset = create_dataset(config, self.data_root, VAL)
         label_mappings = train_dataset.label_mappings
-        if self.config.system.dryrun:
-            train_dataset = create_dryrun_dataset(config, train_dataset, TRAIN)
```

Comment:
Why are you removing dry run? if we are not supporting dry run why we have create_dryrun_dataset method in this class?

Reply:
@adason Can you explain why we are removing dryrun. Also you are right the create_dryrun_dataset could be deleted if this is correct. I will wait for YC to respond

Reply:
We should be able to use DummyDetecction2D dataset for the same purpose of dry-run. The original implementation still require user to download the whole dataset in order to run train/evaluate. This does not help development process.  We will also not supply `--dry-run` in cli.

Reply:
WIll delete the function create_dryrun_dataset